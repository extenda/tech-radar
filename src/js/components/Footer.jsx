import React from 'react';
import radarService from '../modules/radarService';
import Icon from './Icon';

const Dot = () => (
  <React.Fragment>
    {' '}
    &#8729;
    {' '}
  </React.Fragment>
);

const Footer = () => {
  const { title, version, formattedDate } = radarService.model;
  return (
    <footer className="footer">
      <div className="container">
        <p className="footer-text">
          {title}
          {' '}
          {version}
          {', '}
          {formattedDate}
          <Dot />
          <Icon name="copyright" spaceBefore={false} />
          <a href="https://extendaretail.com">Extenda Retail</a>
          <br />
          Maintained at
          <Icon name="github" />
          <a href="https://github.com/extenda/tech-radar">GitHub</a>
          <Dot />
          Discuss in
          <Icon name="slack" />
          <a href="https://extendaretail.slack.com/channels/tech-radar">tech-radar</a>
          <br />
          The Tech Radar Generator is licensed under
          {' '}
          <a href="https://github.com/extenda/tech-radar/blob/master/LICENSE">MIT</a>
          <Dot />
          Thanks to Zalando for the
          {' '}
          <a href="https://github.com/zalando/tech-radar">radar visualization</a>
        </p>
      </div>
    </footer>
  );
};

export default Footer;
