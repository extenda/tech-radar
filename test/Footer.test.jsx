import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../src/js/components/Footer';

jest.mock('../src/js/modules/radarService');

describe('<Footer />', () => {
  test('It renders the footer', () => {
    const footer = shallow(<Footer />);
    expect(footer).toMatchSnapshot();
  });
});
