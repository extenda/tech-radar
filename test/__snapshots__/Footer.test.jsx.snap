// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> It renders the footer 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <p
      className="footer-text"
    >
      Test 0.0.1-local, 5 June, 2019
       ∙ 
      <Icon
        name="copyright"
        spaceBefore={false}
      />
      <a
        href="https://extendaretail.com"
      >
        Extenda Retail
      </a>
      <br />
      Maintained at
      <Icon
        name="github"
        spaceBefore={true}
      />
      <a
        href="https://github.com/extenda/tech-radar"
      >
        GitHub
      </a>
       ∙ 
      Discuss in
      <Icon
        name="slack"
        spaceBefore={true}
      />
      <a
        href="https://extendaretail.slack.com/channels/tech-radar"
      >
        tech-radar
      </a>
      <br />
      The Tech Radar Generator is licensed under
       
      <a
        href="https://github.com/extenda/tech-radar/blob/master/LICENSE"
      >
        MIT
      </a>
       ∙ 
      Thanks to Zalando for the
       
      <a
        href="https://github.com/zalando/tech-radar"
      >
        radar visualization
      </a>
    </p>
  </div>
</footer>
`;
